### ISSUE TITLE
kubecon CFP

### ISSUE THREAD

**Issue by Dieterbe** [grafana/metrictank#1589]
Let's see if we can gather enough things about our experience running metrictank on kubernetes
If we do, we can submit a proposal to kubecon. I think it would be interesting to people if we just share "lessons learned / random tips"

we can talk about:
* the issue on GKE with cordon/taint
* interesting things when working with the k8s api (I know @fkaleo has some recent complaints)
* we have to make many unique deployments because of our assigned partition scheme, meaning we can have to re-invent a couple things (like rollout concurrency)
* cpu limits problem
* changes to mounted config maps don't trigger redeploys, need manual tweaks
* sometimes we've had to manually create or rename deployments. there has been some trickery here, making sure the new deployment doesn't impact the old one, etc. Also for our own sake, it would be nice if we explained this better.

lessons learned wrt upgrades? @jimnagy ?

**Comment by jimnagy** [comment]
The main thing I learned during the first round of upgrades was the gke cordon/taint issue, which you've already mentioned.  There are some github issues that discuss it, but it can't hurt to make more people aware of it.

Could also discuss the somewhat unique factors about metrictank that factored in why we went with our own automated upgrade process.  Basically, the length of time it takes for pods to come up and the  desire to not have a ton of pods replaying from kafka at the same time.

I'll go through my old notes from the upgrades and see if I can come up with anything else.
